src/CattyTests/Download/StoreProjectsDownloaderTests.swift

```diff
@@ -575,6 +575,84 @@ class StoreProjectsDownloaderTests: XCTestCase {
     }
 
+    // MARK: - Download project
+    func testDownloadProjectSucceeds() {
+           let dvrSession = Session(cassetteName: "StoreProjectDownloader.downloadProject.success")
+           let downloader = StoreProjectDownloader(session: dvrSession)
+           let project = getStoreProjectMock()
+           let expectation = XCTestExpectation(description: "Download Project")
+
+        downloader.download(project: project, completion: { data, error in
+            XCTAssertNil(error, "request failed")
+            guard data != nil else { XCTFail("no data received"); return }
+            expectation.fulfill()
+        }, progression: nil)
+
+            wait(for: [expectation], timeout: 1)
+    }
+
+    func testDownloadProjectFailsWithUnexpectedError() {
+        let mockSession = URLSessionMock()
+        let downloader = StoreProjectDownloader(session: mockSession)
+        let expectation = XCTestExpectation(description: "Download Projects")
+        let project = getStoreProjectMock()
+
+        downloader.download(project: project, completion: {_, error in
+            guard let error = error else { XCTFail("no error returned"); return }
+            XCTAssertEqual(error, .unexpectedError)
+            expectation.fulfill()
+        }, progression: nil)
+
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testDownloadprojectFailsWithRequestError() {
+        let dvrSession = Session(cassetteName: "StoreProjectDownloader.downloadProject.fail.request")
+        let downloader = StoreProjectDownloader(session: dvrSession)
+        let expectation = XCTestExpectation(description: "Download Projects")
+        let project = getStoreProjectMock()
+
+        downloader.download(project: project, completion: { _, error in
+            guard let error = error else { XCTFail("no error received"); return }
+            switch error {
+            case let .request(error: _, statusCode: statusCode):
+                XCTAssertEqual(statusCode, 404)
+            default:
+                XCTFail("wrong error received")
+            }
+            expectation.fulfill()
+        }, progression: nil)
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testDownloadProjectProgression() {
+        let config = URLSessionConfiguration.ephemeral
+        let session = URLSession(configuration: config, delegate: nil, delegateQueue: nil)
+        let downloader = StoreProjectDownloader(session: session)
+        let expectation = XCTestExpectation(description: "Download Project Progression")
+        let project = getStoreProjectMock()
+
+        downloader.download(project: project, completion: { _, _ in }) { progress in
+            XCTAssertTrue(!progress.isNaN)
+            expectation.fulfill()
+        }
+         wait(for: [expectation], timeout: 5)
+    }
+
+    func testDownloadProjectVaildProgression() {
```

Comment:
Is there a difference between this and the test above? Please have a look at the typo in your method name. Nice documentation that the progress value is actually between 0 and 1 (maybe you can also add XCTAssertTrue(progress > 0)).